### PR TITLE
feat(lease.sh): Add cloud provider support

### PR DESCRIPTION
**Pull request by jchauncey**
This adds the ability to specify the cloud provider when starting an e2e run. This will be defaulted in the jenkins-jobs repo to google for now.

### CODE REVIEW

**Comment by arschles** [comment]
LGTM, but not sure about CI @jchauncey.

Also, does this require https://github.com/deis/k8s-claimer/pull/104 to be merged & deployed?

**Comment by jchauncey** [comment]
Yes we will need the k8s claimer PR merged before this one can be merged

**Comment by jchauncey** [comment]
This will need a change before we merge it which is to update the Dockerfile with the link to the new client binary

File: Dockerfile [hunk 1]
```diff
@@ -21,7 +21,7 @@ RUN apt-get update -y && \
 	apt-get install -y \
 		git \
 		unzip && \
-	curl -Ls -o /usr/bin/k8s-claimer https://storage.googleapis.com/k8s-claimer/git-e4dcc16/k8s-claimer-git-e4dcc16-linux-amd64 && \
```
Comment (sgoings) [inline Dockerfile:24]: What's the reasoning behind moving this from an immutable version to a mutable one? I'm not sure I'm a fan of the potential increased surface for future failz.

File: Dockerfile [hunk 2]
```diff
@@ -21,7 +21,7 @@ RUN apt-get update -y && \
...
+	curl -Ls -o /usr/bin/k8s-claimer https://k8sclaimercli.blob.core.windows.net/cli/k8s-claimer-latest-linux-amd64 && \
```
Comment (arschles) [inline Dockerfile]: I missed this before, but realized from what @sgoings said that I think this should probably be versioned @jchauncey

**Review by arschles — COMMENTED** [review]
@jchauncey only one comment about the mutability of `kubectl`

File: Dockerfile (hunk 2 above)
Comment (jchauncey) [reply Dockerfile]: You would still need to trigger a rebuild of hte image to fetch the latest and its done by hand or through a PR to e2e-runner. We can change it back but just felt like this makes it a little easier for us to roll out changes wihtout having to touch a bunch of different repos.

Comment (jchauncey) [reply Dockerfile]: im going to change this back

**Pull request merged**